apps/obsidian/src/components/Settings.tsx

```diff
@@ -1,181 +1,92 @@
+import { ContextProvider } from "./AppContext";
+import RelationshipTypeSettings from "./RelationshipTypeSettings";
+import RelationshipSettings from "./RelationshipSettings";
+import NodeTypeSettings from "./NodeTypeSettings";
+import { PluginProvider, usePlugin } from "./PluginContext";
 
-const NodeTypeSettings = ({ plugin }: { plugin: DiscourseGraphPlugin }) => {
-  const [nodeTypes, setNodeTypes] = useState(
-    () => plugin.settings.nodeTypes ?? [],
-  );
-  const [formatErrors, setFormatErrors] = useState<Record<number, string>>({});
-  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
-
-  useEffect(() => {
-    const initializeSettings = async () => {
-      let needsSave = false;
-
-      if (!plugin.settings.nodeTypes) {
-        plugin.settings.nodeTypes = [];
-        needsSave = true;
-      }
-
-      if (needsSave) {
-        await plugin.saveSettings();
-      }
-    };
-
-    initializeSettings();
-  }, [plugin]);
-
-  const handleNodeTypeChange = async (
-    index: number,
-    field: "name" | "format",
-    value: string,
-  ): Promise<void> => {
-    const updatedNodeTypes = [...nodeTypes];
-    if (!updatedNodeTypes[index]) {
-      updatedNodeTypes[index] = { name: "", format: "" };
-    }
-
-    updatedNodeTypes[index][field] = value;
-    setNodeTypes(updatedNodeTypes);
-    setHasUnsavedChanges(true);
-
-    if (field === "format") {
-      const { isValid, error } = validateNodeFormat(value);
-      if (!isValid) {
-        setFormatErrors((prev) => ({
-          ...prev,
-          [index]: error ?? "Invalid format",
-        }));
-      } else {
-        setFormatErrors((prev) => {
-          const newErrors = { ...prev };
-          delete newErrors[index];
-          return newErrors;
-        });
-      }
-    }
-  };
-
-  const handleAddNodeType = (): void => {
-    const updatedNodeTypes = [
-      ...nodeTypes,
-      {
-        name: "",
-        format: "",
-      },
-    ];
-    setNodeTypes(updatedNodeTypes);
-    setHasUnsavedChanges(true);
-  };
-
-  const handleDeleteNodeType = async (index: number): Promise<void> => {
-    const updatedNodeTypes = nodeTypes.filter((_, i) => i !== index);
-    setNodeTypes(updatedNodeTypes);
-    plugin.settings.nodeTypes = updatedNodeTypes;
-    await plugin.saveSettings();
-  };
-
-  const handleSave = async (): Promise<void> => {
-    let hasErrors = false;
-    for (let i = 0; i < nodeTypes.length; i++) {
-      const { isValid, error } = validateNodeFormat(nodeTypes[i]?.format ?? "");
-      if (!isValid) {
-        setFormatErrors((prev) => ({
-          ...prev,
-          [i]: error ?? "Invalid format",
-        }));
-        hasErrors = true;
-      }
-    }
-
-    if (hasErrors) {
-      return;
-    }
-
-    plugin.settings.nodeTypes = nodeTypes;
-    await plugin.saveSettings();
-    setHasUnsavedChanges(false);
-  };
+const Settings = () => {
+  const plugin = usePlugin();
+  const [activeTab, setActiveTab] = useState("nodeTypes");
```

Comment:
nit - we can address this later
since we are using the unsaved changes pattern, we should probably also warn before tab switch if unsaved changes